packages/Sandblocks-Core/SBForceMoveDecorator.class.st

```diff
@@ -199,18 +242,58 @@ SBForceMoveDecorator >> isPinned [
 ]
 
 { #category : #'as yet unclassified' }
-SBForceMoveDecorator >> mouseUp: anEvent [
+SBForceMoveDecorator >> isPinned: aBoolean [
+
+	self pinnedPosition: (aBoolean ifTrue: [self morph position] ifFalse: [nil])
+]
+
+{ #category : #'as yet unclassified' }
+SBForceMoveDecorator >> morphAtPoint: aPoint satisfying: aBlock [
+
+	| best bestArea |
+	best := nil.
+	bestArea := 0.
 	
-	self isDragging ifTrue: [ | otherBlock |
+	self morph owner submorphsDo: [:aMorph |
+		((aMorph containsPoint: aPoint)
+			and: [aMorph bounds area > bestArea]
+			and: [aMorph ~= self morph]
+			and: [aBlock value: aMorph]) ifTrue: [
+			best := aMorph.
+			bestArea := aMorph bounds area]].
+	
+	^ best
+]
+
+{ #category : #'as yet unclassified' }
+SBForceMoveDecorator >> mouseUp: anEvent [
+
+	self isDragging ifTrue: [ | otherBlock otherMorph |
 		self idealPosition: self morph position.
 		self pinnedPosition ifNotNil: [self pinnedPosition: self morph position].
```

Comment:
could it make sense to move updating the pinned position into the idealPosition setter or a different setter?

Reply:
to explain the rationale: it appears like something that would maybe want to be updated together with another position value in multiple spots. I haven't checked though, so if it's only ever the user overriding it via mouseUp then it's already perfect :)